src/aatoolbox/datasources/usgs/usgs_ndvi.py

```diff
@@ -85,117 +75,85 @@ class _UsgsNdvi(DataSource):
-    end_date : Union[date, str, Tuple[int], None]
+        object or a data string in ISO8601 format, and
+        the relevant dekad will be determined. Or pass
+        directly as year-dekad tuple, e.g. (2020, 1).
+        If ``None``, ``start_date`` is set to earliest
+        date with data: 2002, dekad 19.
+    end_date : _DATE_TYPE, default = None
         End date. Can be passed as a ``datetime.date``
         object and the relevant dekad will be determined,
         as a date string in ISO8601 format, or as a
-        year-dekad tuple, i.e. (2020, 1).
-    processed_file_suffix : str
+        year-dekad tuple, i.e. (2020, 1). If ``None``,
+        ``end_date`` is set to ``date.today()``.
+    processed_file_suffix : Optional[str], default = None
         Suffix for processed file.
     """
 
     def __init__(
         self,
         country_config: CountryConfig,
-        coverage_area: _VALID_AREAS,
-        data_type: _VALID_TYPES,
-        start_date: Union[date, str, Tuple[int], None] = None,
-        end_date: Union[date, str, Tuple[int], None] = None,
-        processed_file_suffix: str = "",
+        data_variable: str,
+        data_variable_suffix: str,
+        data_variable_url: str,
+        start_date: _DATE_TYPE = None,
+        end_date: _DATE_TYPE = None,
+        processed_file_suffix: Optional[str] = None,
     ):
         super().__init__(
             country_config=country_config,
             datasource_base_dir="usgs_ndvi",
             is_public=True,
             is_global_raw=True,
         )
-        self.coverage_area = coverage_area
-
-        # set data type directly in init because should not be
-        # changed in an instance
-        valid_types = _VALID_TYPES.__args__  # type: ignore
-        if data_type not in valid_types:
-            raise ValueError(
-                "`data_type` is not a valid string value. "
-                f"It must be one of {*valid_types,}"
+
+        # set area url and prefix from config
+        if self._country_config.usgs_ndvi is None:
+            raise AttributeError(
+                "The country configuration file does not contain "
+                "any USGS NDVI area name. Please update the config file and "
+                "try again. See the documentation for the valid area names."
             )
-        self._data_type = data_type
-
-        # url code and file suffix
-        loc_data_types = {
-            "smoothed": ("temporallysmoothedndvi", ""),
-            "percent_median": ("percentofmedian", "pct"),
-            "median_anomaly": ("mediananomaly", "stmdn"),
-            "difference": ("differencepreviousyear", "dif"),
-        }
-        data_type_values = loc_data_types[data_type]
-        self._data_type_url = data_type_values[0]
-        self._data_type_suffix = data_type_values[1]
+        self._area_url = self._country_config.usgs_ndvi.area_url
+        self._area_prefix = self._country_config.usgs_ndvi.area_prefix
```

Comment:
sorry, this wasnt updated yet in the FN pr where I guess you got this from, but now it is (looks like you didn't merge the latest `develop` yet?). Instead of `self._country_config.usgs_ndvi.area_url` you can now use `self._datasource_config.area_url` . And then here, you don't have to assign it anymore to a new var `self._area_url`, but just use the `self._datasource_config.area_url` when you need it. Reason we were assigning it to a new var before is because otherwise mypy would complain